app/pulp/app/models/task.py

```diff
@@ -54,10 +54,12 @@ def get_unreserved_worker(self):
                            entries associated with it.
         :rtype:            pulp.app.models.Worker
         """
-        free_workers_qs = self.annotate(models.Count('reservations')).filter(reservations__count=0)
+        workers_only_qs = Worker.objects.filter(name__startswith='reserved')
+        workers_only_qs_with_counts = workers_only_qs.annotate(models.Count('reservations'))
+        free_workers_qs = workers_only_qs_with_counts.filter(reservations__count=0)
         if free_workers_qs.count() == 0:
```

Comment:
Since you have me thinking about race conditions... :) is it possible there's one here?

Does ``free_workers_qs`` get fully evaluated here, with results loaded into RAM? Or (I suspect this is the case) does it only get the count from the DB, and does a separate query two lines below.

What if the reservation counts change in the mean time? Or a worker goes away entirely?

Perhaps this could instead follow the classic python try...except pattern, which would also eliminate the need for a ``count()`` call. Something like:

```
try:
  return workers_only_qs_with_counts.filter(reservations__count=0).order_by('?')[0]
except IndexError:
  raise self.model.DoesNotExist()
```